src/cps/search.cpp

```diff
@@ -366,13 +367,21 @@ namespace cps::search {
 
         result.version = root->data.package.version.value_or("unknown");
 
+        const auto prefix_path = [&]() -> std::optional<fs::path> {
+            if (prefix_variable) {
+                return fs::path{prefix_variable.value()};
+            }
+            return std::nullopt;
+        }();
+
         for (auto && node : flat) {
 
             const auto && prefix_replacer = [&](const std::string & s) -> std::string {
                 // TODO: Windows…
                 auto && split = utils::split(s, "/");
                 if (split[0] == "@prefix@") {
-                    fs::path p = calculate_prefix(node->data.package.cps_path);
+                    auto p =
+                        prefix_path.has_value() ? prefix_path.value() : calculate_prefix(node->data.package.cps_path);
```

Comment:
Could this be:
```C++
auto p = prefix_path.value_or(calculate_prefix(node->data.package.cps_path));
```
Or would that cause eager evaluation that the ternary avoids?

Reply:
I don't know if it would but I like the style of that better, it is easier to read.